### PR TITLE
Set fd for file resource data to 0 after it has been closed

### LINKED ISSUE #6222: sys.reboot() fails with an exception in HTML5 builds
**Describe the bug**
Calling sys.reboot() on HTML5 causes an exception

**To Reproduce**

1. Add a GUI component to an object in the main collection, with the following input handler, and acquire input focus:
```
function on_input(self, action_id, action)
	if action_id == hash("touch") and action.pressed then
		sys.reboot()
	end
end
```
2. Project → Build HTML5
3. Open Chrome Developer Tools
4. Click in the game window

**Expected behavior**
The game/app restarts

**Defold version:**
  - Version 1.2.189

**Platforms:**
 - Platforms: HTML5
 - OS: Windows 10, Chrome 96

**Logs:**
```
dmloader.js:512 exception thrown: RuntimeError: null function or function signature mismatch,RuntimeError: null function or function signature mismatch
at wasm://wasm/009fec5e:wasm-function[547]:0x9c42
at wasm://wasm/009fec5e:wasm-function[1460]:0x5c69e
at wasm://wasm/009fec5e:wasm-function[3483]:0x11f377
at wasm://wasm/009fec5e:wasm-function[1491]:0x61a8c
at wasm://wasm/009fec5e:wasm-function[3466]:0x11e27c
at wasm://wasm/009fec5e:wasm-function[3502]:0x120941
at wasm://wasm/009fec5e:wasm-function[2200]:0xb8684
at wasm://wasm/009fec5e:wasm-function[4742]:0x18549a
at wasm://wasm/009fec5e:wasm-function[4747]:0x1861ae
at browserIterationFunc (:1:190338)
```

### PR DESCRIPTION
**Pull request by britzl**
Fixes #6222

### CODE REVIEW

File: engine/resource/src/mount/mount_generic.cpp [hunk 1]
```diff
@@ -60,6 +60,7 @@ namespace dmResource
         if (afi && afi->m_FileResourceData)
         {
             fclose(afi->m_FileResourceData);
+            afi->m_FileResourceData = 0;
```
Comment (britzl) [inline engine/resource/src/mount/mount_generic.cpp:63]: UnmountArchiveInternal is called during reboot and here we close `afi->m_FileResourceData`. This function in turn calls dmResourceArchive::Delete(archive) and there we also try to close the same file descriptor.

Comment (JCash) [reply engine/resource/src/mount/mount_generic.cpp:63]: Then I wonder who should own/close it?
I don't think we should have a second case of "if (fd) clos(fd)" if it is never going to be run?

Comment (britzl) [reply engine/resource/src/mount/mount_generic.cpp]: Good point. mount_generic.cpp does not own that file descriptor and should not attempt to close it. Updated the PR!

**Review by JCash — APPROVED** [review]
Looks good!

**Pull request merged**